sql/src/main/scala/org/apache/spark/sql/sedona_sql/expressions/Functions.scala

```diff
@@ -1167,3 +1167,46 @@ case class ST_SubDivideExplode(children: Seq[Expression]) extends Generator {
 
   override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = ev
 }
+
+
+case class ST_MakePolygon(inputExpressions: Seq[Expression])
+  extends Expression with CodegenFallback {
+  override def nullable: Boolean = true
+  private val geometryFactory = new GeometryFactory()
+
+  override def eval(input: InternalRow): Any = {
+    inputExpressions.betweenLength(1, 2)
+    val exteriorRing = inputExpressions.head
+    val possibleHolesRaw = inputExpressions.tail.headOption.map(_.eval(input).asInstanceOf[ArrayData])
+    val numOfElements = possibleHolesRaw.map(_.numElements()).getOrElse(0)
+
+    val holes = (0 until numOfElements).map(el => possibleHolesRaw match {
+      case Some(value) => Some(value.getArray(el))
+      case None => None
+    }).filter(_.nonEmpty)
+      .map(el => el.map(_.toGeometry))
+      .flatMap{
+        case maybeLine: Option[LineString] =>
+          maybeLine.map(line => geometryFactory.createLinearRing(line.getCoordinates))
+        case _ => None
+      }
+
+    exteriorRing.toGeometry(input) match {
+      case geom: LineString =>
+        try {
+          val poly = new Polygon(geometryFactory.createLinearRing(geom.getCoordinates), holes.toArray, geometryFactory)
+          poly.toGenericArrayData
+        }
+        catch {
+          case e: Exception => null
```

Comment:
Will it be better if you throw an exception here instead of giving a null?

Reply:
The reason why I changed to null instead of raising exception was to make that fault tollerant. I can change that, but I heard some complaints about that in some cases application is running for several minutes and then fail due to data quality.

Reply:
Sounds good to me!